### ISSUE TITLE
error installing 

### ISSUE THREAD

**Issue by morganherlocker** [mick/geojson-preview#4]
I get an error when trying to install the package:

> runas@0.5.4 install /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/pathwatcher/node_modules/runas
> node-gyp rebuild
> 
> npm http GET https://registry.npmjs.org/leaflet
> npm http GET https://registry.npmjs.org/geojsonhint
> npm http GET https://registry.npmjs.org/fs-plus
> npm http GET https://registry.npmjs.org/pathwatcher/0.16.0
> npm http GET https://registry.npmjs.org/underscore-plus
> npm http 304 https://registry.npmjs.org/geojsonhint
> npm http 304 https://registry.npmjs.org/fs-plus
> npm http 304 https://registry.npmjs.org/pathwatcher/0.16.0
> npm http 304 https://registry.npmjs.org/leaflet
> npm http 304 https://registry.npmjs.org/underscore-plus
> npm http GET https://registry.npmjs.org/underscore
> npm http GET https://registry.npmjs.org/mkdirp
> npm http GET https://registry.npmjs.org/rimraf
> npm http GET https://registry.npmjs.org/async
> npm http GET https://registry.npmjs.org/bindings
> npm http GET https://registry.npmjs.org/nan/0.8.0
> npm http GET https://registry.npmjs.org/q
> npm http GET https://registry.npmjs.org/emissary
> npm http GET https://registry.npmjs.org/async
> npm http GET https://registry.npmjs.org/runas
> npm http GET https://registry.npmjs.org/jsonlint-lines
> npm http GET https://registry.npmjs.org/optimist
> npm http GET https://registry.npmjs.org/colors
> npm http 304 https://registry.npmjs.org/async
> npm http 304 https://registry.npmjs.org/mkdirp
> npm http 304 https://registry.npmjs.org/async
> npm http 304 https://registry.npmjs.org/bindings
> npm http 304 https://registry.npmjs.org/rimraf
> npm http 304 https://registry.npmjs.org/underscore
> npm http 304 https://registry.npmjs.org/emissary
> npm http 304 https://registry.npmjs.org/q
> npm http 304 https://registry.npmjs.org/nan/0.8.0
> npm http 304 https://registry.npmjs.org/runas
> npm http 304 https://registry.npmjs.org/colors
> npm http 304 https://registry.npmjs.org/jsonlint-lines
> /Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/node_modules/npm/bin/node-gyp-bin/node-gyp: line 2: node: command not found
> npm http GET https://registry.npmjs.org/mixto
> npm http GET https://registry.npmjs.org/property-accessors
> npm http 304 https://registry.npmjs.org/optimist
> npm ERR! Error: ENOENT, lstat '/private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/pathwatcher/node_modules/q/q.js'
> npm ERR! If you need help, you may report this _entire_ log,
> npm ERR! including the npm and node versions, at:
> npm ERR!     http://github.com/isaacs/npm/issues
> 
> npm ERR! System Darwin 12.5.0
> npm ERR! command "/Applications/Atom.app/Contents/Frameworks/Atom Helper.app/Contents/MacOS/Atom Helper" "/Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/node_modules/npm/bin/npm-cli.js" "--userconfig" "/Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/.apmrc" "install" "/Users/morgan/.atom/.node-gyp/.atom/.apm/geojson-preview/0.2.0/package.tgz" "--target=0.11.10" "--arch=x64"
> npm ERR! cwd /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze
> npm ERR! node -v v0.11.10
> npm ERR! npm -v 1.3.18
> npm ERR! path /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/pathwatcher/node_modules/q/q.js
> npm ERR! fstream_path /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/pathwatcher/node_modules/q/q.js
> npm ERR! fstream_type File
> npm ERR! fstream_class FileWriter
> npm ERR! code ENOENT
> npm ERR! errno -2
> npm ERR! fstream_stack /Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/node_modules/npm/node_modules/fstream/lib/writer.js:284:26
> npm ERR! fstream_stack Object.oncomplete (fs.js:97:15)
> npm ERR! Error: ENOENT, lstat '/private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/pathwatcher/node_modules/nan/nan.h'
> npm ERR! If you need help, you may report this _entire_ log,
> npm ERR! including the npm and node versions, at:
> npm ERR!     http://github.com/isaacs/npm/issues
> 
> npm ERR! System Darwin 12.5.0
> npm ERR! command "/Applications/Atom.app/Contents/Frameworks/Atom Helper.app/Contents/MacOS/Atom Helper" "/Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/node_modules/npm/bin/npm-cli.js" "--userconfig" "/Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/.apmrc" "install" "/Users/morgan/.atom/.node-gyp/.atom/.apm/geojson-preview/0.2.0/package.tgz" "--target=0.11.10" "--arch=x64"
> npm ERR! cwd /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze
> npm ERR! node -v v0.11.10
> npm ERR! npm -v 1.3.18
> npm ERR! path /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/pathwatcher/node_modules/nan/nan.h
> npm ERR! fstream_path /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/pathwatcher/node_modules/nan/nan.h
> npm ERR! fstream_type File
> npm ERR! fstream_class FileWriter
> npm ERR! code ENOENT
> npm ERR! errno -2
> npm ERR! fstream_stack /Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/node_modules/npm/node_modules/fstream/lib/writer.js:284:26
> npm ERR! fstream_stack Object.oncomplete (fs.js:97:15)
> npm ERR! Error: ENOENT, lstat '/private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/leaflet/debug/vector/geojson-sample.js'
> npm ERR! If you need help, you may report this _entire_ log,
> npm ERR! including the npm and node versions, at:
> npm ERR!     http://github.com/isaacs/npm/issues
> 
> npm ERR! System Darwin 12.5.0
> npm ERR! command "/Applications/Atom.app/Contents/Frameworks/Atom Helper.app/Contents/MacOS/Atom Helper" "/Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/node_modules/npm/bin/npm-cli.js" "--userconfig" "/Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/.apmrc" "install" "/Users/morgan/.atom/.node-gyp/.atom/.apm/geojson-preview/0.2.0/package.tgz" "--target=0.11.10" "--arch=x64"
> npm ERR! cwd /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze
> npm ERR! node -v v0.11.10
> npm ERR! npm -v 1.3.18
> npm ERR! path /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/leaflet/debug/vector/geojson-sample.js
> npm ERR! fstream_path /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/leaflet/debug/vector/geojson-sample.js
> npm ERR! fstream_type File
> npm ERR! fstream_class FileWriter
> npm ERR! code ENOENT
> npm ERR! errno -2
> npm ERR! fstream_stack /Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/node_modules/npm/node_modules/fstream/lib/writer.js:284:26
> npm ERR! fstream_stack Object.oncomplete (fs.js:97:15)
> npm ERR! error rolling back Error: ENOTEMPTY, rmdir '/private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/geojsonhint'
> npm ERR! error rolling back  geojson-preview@0.2.0 { [Error: ENOTEMPTY, rmdir '/private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/geojsonhint']
> npm ERR! error rolling back   stack: 'Error: ENOTEMPTY, rmdir \'/private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/geojsonhint\'',
> npm ERR! error rolling back   errno: -66,
> npm ERR! error rolling back   code: 'ENOTEMPTY',
> 
> npm ERR! error rolling back   path: '/private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/geojsonhint' }
> npm ERR! runas@0.5.4 install: `node-gyp rebuild`
> npm ERR! Exit status 127
> npm ERR! 
> npm ERR! Failed at the runas@0.5.4 install script.
> npm ERR! This is most likely a problem with the runas package,
> npm ERR! not with npm itself.
> npm ERR! Tell the author that this fails on your system:
> npm ERR!     node-gyp rebuild
> npm ERR! You can get their info via:
> npm ERR!     npm owner ls runas
> npm ERR! There is likely additional logging output above.
> 
> npm ERR! System Darwin 12.5.0
> npm ERR! command "/Applications/Atom.app/Contents/Frameworks/Atom Helper.app/Contents/MacOS/Atom Helper" "/Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/node_modules/npm/bin/npm-cli.js" "--userconfig" "/Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/.apmrc" "install" "/Users/morgan/.atom/.node-gyp/.atom/.apm/geojson-preview/0.2.0/package.tgz" "--target=0.11.10" "--arch=x64"
> npm ERR! cwd /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze
> npm ERR! node -v v0.11.10
> npm ERR! npm -v 1.3.18
> npm ERR! code ELIFECYCLE
> npm ERR! Error: ENOENT, lstat '/private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/pathwatcher/node_modules/async/README.md'
> npm ERR! If you need help, you may report this _entire_ log,
> npm ERR! including the npm and node versions, at:
> npm ERR!     http://github.com/isaacs/npm/issues
> 
> npm ERR! System Darwin 12.5.0
> npm ERR! command "/Applications/Atom.app/Contents/Frameworks/Atom Helper.app/Contents/MacOS/Atom Helper" "/Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/node_modules/npm/bin/npm-cli.js" "--userconfig" "/Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/.apmrc" "install" "/Users/morgan/.atom/.node-gyp/.atom/.apm/geojson-preview/0.2.0/package.tgz" "--target=0.11.10" "--arch=x64"
> npm ERR! cwd /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze
> npm ERR! node -v v0.11.10
> npm ERR! npm -v 1.3.18
> npm ERR! path /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/pathwatcher/node_modules/async/README.md
> npm ERR! fstream_path /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/node_modules/geojson-preview/node_modules/pathwatcher/node_modules/async/README.md
> npm ERR! fstream_type File
> npm ERR! fstream_class FileWriter
> npm ERR! code ENOENT
> npm ERR! errno -2
> npm ERR! fstream_stack /Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/node_modules/npm/node_modules/fstream/lib/writer.js:284:26
> npm ERR! fstream_stack Object.oncomplete (fs.js:97:15)
> npm http GET https://registry.npmjs.org/wordwrap
> npm http GET https://registry.npmjs.org/minimist
> npm http GET https://registry.npmjs.org/nomnom
> npm http GET https://registry.npmjs.org/JSV
> npm http 304 https://registry.npmjs.org/mixto
> npm http 304 https://registry.npmjs.org/property-accessors
> npm http 304 https://registry.npmjs.org/wordwrap
> npm http 304 https://registry.npmjs.org/minimist
> npm http 304 https://registry.npmjs.org/JSV
> npm http 304 https://registry.npmjs.org/nomnom
> npm ERR! 
> npm ERR! Additional logging details can be found in:
> npm ERR!     /private/var/folders/5b/1n6ylh2d1g38bvpqqk22pc080000gn/T/apm-install-dir-11422-76978-xaenze/npm-debug.log
> npm ERR! not ok code 0

**Comment by morganherlocker** [comment]
Install is successful from the command line, but not from the settings interface.

**Comment by mick** [comment]
I get the same thing when installing from inside of atom. 

Looks like the same as: https://github.com/atom/markdown-preview/issues/31 which seems to be a problem running node-gyp, which pathwatcher needs to compile a dependency.

**Comment by kevinsawicki** [comment]
Hi, can you confirm whether you are still seeing this after upgrading to Atom 0.65.0.

That version of Atom comes bundled with node so these issues where node-gyp can't find node should be resolved.

**Comment by mick** [comment]
@kevinsawicki Hey! I just upgraded to 0.66.0 And I get the same error [gist](https://gist.github.com/mick/9358819)

Any other info I can provide to help debug this?

**Comment by kevinsawicki** [comment]
@mick can you run `apm -v` and provide the output?

**Comment by mick** [comment]
@kevinsawicki  here it is:

```
 $ apm -v
 apm  0.26.0
 npm  1.4.4
 node 0.10.26
```

**Comment by kevinsawicki** [comment]
Thanks, taking another look.

**Comment by kevinsawicki** [comment]
And do you have node installed? `which node`

**Comment by mick** [comment]
Yep

```
$ which node
/usr/local/bin/node
$ node -v
v0.10.26
```

**Comment by kevinsawicki** [comment]
There are a lot of errors in this log but I was assuming this was the culprit:

`/Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/node_modules/npm/bin/node-gyp-bin/node-gyp: line 2: node: command not found`

So I'm going to investigate that first, making sure node is on the `PATH` when invoked from the settings view.

**Comment by mick** [comment]
@kevinsawicki I agree, it looks like the other errors are a result of that failing.  Thanks for looking into this!

**Comment by kevinsawicki** [comment]
Okay, I think I might have a fix, can you copy the file here: https://gist.github.com/kevinsawicki/c0895a12a2450d56d6c1

to `/Applications/Atom.app/Contents/Resources/app/node_modules/atom-package-manager/lib/install.js` and try again?

**Comment by mick** [comment]
:+1: that works!

**Comment by kevinsawicki** [comment]
Awesome, thanks so much for trying it out, will try to get a new release out soon with the fix.

**Comment by mick** [comment]
Great! I'll keep an eye out for the new release.  Thanks for look into this.

**Comment by kevinsawicki** [comment]
Okay, Atom 0.67.0 is out with this fix, you should see an update roll out in the next hour or so.

Thanks again for helping me track this down, much appreciated :steam_locomotive::dash:

**Comment by mick** [comment]
@kevinsawicki btw, is there a way to force atom to check for updates?

**Comment by kevinsawicki** [comment]
There should be an Atom > Check for Update menu, you may need to wait a couple seconds for it to show after launching Atom.

**Comment by mick** [comment]
This is what I have:
![screen shot 2014-03-05 at 12 31 05 pm](https://f.cloud.github.com/assets/26278/2338080/37b29c54-a4a5-11e3-8796-1c4aa8fb8af5.png)

Its been running since before you published 0.67.0 yesterday.

**Comment by mick** [comment]
It did appear when I restarted again. Anyways unrelated to this ticket. Thanks again for the fix!
